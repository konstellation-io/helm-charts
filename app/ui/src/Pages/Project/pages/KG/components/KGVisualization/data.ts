// import randomSentence from 'random-sentence';

enum ResourceType {
  CODE = 'code',
  PAPER = 'paper',
}

const data = [
  {
    category: 'Topic A',
    name: 'FaceX-Zoo: A PyTorch Toolbox for Face Recognition',
    type: ResourceType.CODE,
    score: 1,
  },
  {
    category: 'Topic A',
    name: 'RepVGG: Making VGG-style ConvNets Great Again',
    type: ResourceType.PAPER,
    score: 0.9,
  },
  {
    category: 'Topic A',
    name: 'PP-OCR: A Practical Ultra Lightweight OCR System',
    type: ResourceType.PAPER,
    score: 0.91,
  },
  {
    category: 'Topic A',
    name: 'ArtEmis: Affective Language for Visual Art',
    type: ResourceType.CODE,
    score: 0.85,
  },
  {
    category: 'Topic A',
    name: 'Free Lunch for Few-shot Learning: Distribution Calibration',
    type: ResourceType.CODE,
    score: 0.87,
  },
  {
    category: 'Topic A',
    name: 'Reformer: The Efficient Transformer',
    type: ResourceType.PAPER,
    score: 0.7,
  },
  {
    category: 'Topic A',
    name: 'Image Matching across Wide Baselines: From Paper to Practice',
    type: ResourceType.CODE,
    score: 0.65,
  },
  {
    category: 'Topic B',
    name: 'Wasserstein GAN',
    type: ResourceType.CODE,
    score: 0.9,
  },
  {
    category: 'Topic C',
    name: 'Improved Training of Wasserstein GANs',
    type: ResourceType.CODE,
    score: 0.52,
  },
  {
    category: 'Topic D',
    name: 'Scikit-learn: Machine Learning in Python',
    type: ResourceType.CODE,
    score: 0.12,
  },
  {
    category: 'Topic E',
    name: 'Nearest Neighbor Median Shift Clustering for Binary Data',
    type: ResourceType.CODE,
    score: 0.76,
  },
  {
    category: 'Topic F',
    name: 'Path Aggregation Network for Instance Segmentation',
    type: ResourceType.PAPER,
    score: 0.86,
  },
  {
    category: 'Topic G',
    name: 'Refining activation downsampling with SoftPool',
    type: ResourceType.PAPER,
    score: 0.26,
  },
  {
    category: 'Topic G',
    name: 'Scaled-YOLOv4: Scaling Cross Stage Partial Network',
    type: ResourceType.PAPER,
    score: 0.54,
  },
  {
    category: 'Topic G',
    name: 'Real-Time High-Resolution Background Matting',
    type: ResourceType.CODE,
    score: 0.95,
  },
  {
    category: 'Topic H',
    name: 'Scale Equivariance Improves Siamese Tracking',
    type: ResourceType.PAPER,
    score: 0.73,
  },
];

// const categories = [
//   'Topic A',
//   'Topic B',
//   'Topic C',
//   'Topic D',
//   'Topic E',
//   'Topic F',
//   'Topic G',
//   'Topic H',
// ];
// const types = [
//   ResourceType.CODE,
//   ResourceType.PAPER
// ];

// const data = Array(100).fill(1).map(_ => ({
//   category: categories[Math.floor(Math.random() * categories.length)],
//   name: randomSentence({words: 5}),
//   type: types[Math.floor(Math.random() * types.length)],
//   score: Math.random()
// }));

export default data;
