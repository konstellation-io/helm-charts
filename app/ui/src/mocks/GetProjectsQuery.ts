import { project1 } from './entities/project';

export default {
  projects: [project1],
};
