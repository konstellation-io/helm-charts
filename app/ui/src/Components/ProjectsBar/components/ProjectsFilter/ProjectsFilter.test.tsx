import React from 'react';
import ProjectsFilter from './ProjectsFilter';
import GetProjectsQuery from 'Graphql/queries/getProjects';
import { createMockClient } from 'mock-apollo-client';
import { screen, fireEvent, cleanup } from '@testing-library/react';
import {
  apolloRender,
  loadingHandler,
  dataHandler,
  errorHandler,
  getSnapshot,
  ERROR_MESSAGE,
} from 'testUtils';
import data from 'mocks/GetProjectsQuery';

window.HTMLElement.prototype.scrollIntoView = function () {};

let mockClient;
beforeEach(() => {
  mockClient = createMockClient();
});
afterEach(cleanup);

describe('When component is loading', () => {
  beforeEach(() => {
    mockClient.setRequestHandler(GetProjectsQuery, loadingHandler);
    apolloRender(<ProjectsFilter />, mockClient);
  });

  it('should render without crashing', () => {
    expect(screen.getByTestId('spinner')).toBeInTheDocument();
  });
});

describe('When data is ready', () => {
  beforeEach(() => {
    mockClient.setRequestHandler(GetProjectsQuery, dataHandler(data));
    apolloRender(<ProjectsFilter />, mockClient);
  });

  it('should render without crashing', () => {
    expect(screen.getByText('Active', { exact: false })).toBeInTheDocument();
    expect(getSnapshot()).toMatchSnapshot();
  });

  it('should allow the selection of other options', function () {
    // Arrange.
    // Act.
    fireEvent.click(screen.getByText('active', { exact: false }));
    fireEvent.click(screen.getByText('all', { exact: false }));

    // Assert.
    expect(screen.getByText('all', { exact: false })).toBeInTheDocument();
    expect(
      screen.queryByText('active', { exact: false })
    ).not.toBeInTheDocument();
  });
});

describe('When there is an error', () => {
  beforeEach(() => {
    mockClient.setRequestHandler(GetProjectsQuery, errorHandler(''));
    apolloRender(<ProjectsFilter />, mockClient);
  });

  it('should render without crashing', () => {
    expect(screen.getByText(ERROR_MESSAGE)).toBeInTheDocument();
  });
});
