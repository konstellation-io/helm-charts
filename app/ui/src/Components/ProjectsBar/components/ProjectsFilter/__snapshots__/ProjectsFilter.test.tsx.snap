// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When data is ready should render without crashing 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="formSelect _2rBeV _1z7Bn"
      >
        
        <div
          class="_Z0bIQ"
        >
          <div
            class="_3Loey"
            style="height: 67px;"
          >
            Active (1)
          </div>
          <div
            class="_3bqns"
            style="max-height: 0;"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
